### ISSUE TITLE
the regional attention network cannot converge.

### ISSUE THREAD

**Issue by killianMei** [jaeyoon1603/Retrieval-RegionalAttention#4]
I try to run it.Training the regional attention network does not make the training set accurate to 100%. and the network can't converge. What is the reason for this? I use my own training set.
Looking forward to your reply.

**Comment by jaeyoon1603** [comment]
Hi,
It would be happened if parameter update for feature and classification layer is not frozen while training regional attention network.
This is because jointly updating attention layer and other layers is likely to make the training unstable.
Also, note that feature layer and classification layer should be pre-trained.

**Comment by killianMei** [comment]
Hi， @jaeyoon1603 ，I'm very happy to have received your response.
As you said, i have frozen the feature and fc layer, and set the 'requires_grad' to False. 
Firstly, i try to train resnet on my dataset, it shows about 100% accuracy. However, it will drop to 96% accuracy when I join the attention network.This makes me very confused.

**Comment by jaeyoon1603** [comment]
I think it is because regional features are used for classification.
As you can see in the code, the input of fc layer is a summation of regional feature vectors, not a global average pooled vector from feature maps.

So, this kind of setting would decrease the classification accuracy but increase retrieval accuracy.

**Comment by killianMei** [comment]
Glad to receive your reply! Thank you so much. I think I've got it. It was very helpful. Thanks again.
